### PR TITLE
2.x Add a Date/Time guide

### PR DESCRIPTION
**Pull request by gchtr**
**Ticket**: #2104, #2127, #2135

## Issue

It’s not very clear how to work with dates and times in Timber. And we all now, working with dates and times is hard 😅.

## Solution

I added a new guide for working with dates and times in Timber. I’ve figured that writing a Date/Time guide for Timber is actually mostly a guide for working with Date/Time in WordPress itself.

The new guide builds on top of the [updates to the Date/Time component](https://make.wordpress.org/core/2019/09/23/date-time-improvements-wp-5-3/) in WordPress 5.3.

It highlights what you should and shouldn’t do. E.g., you should never use `date_default_timezone_set()`, which is probably the base problem in #2104.

**Help wanted** – In the guide I wrote down what I know about working with dates and times in WordPress. It could be that I’m still wrong in some places and would be glad if someone could help review that information.

## Impact

Better documentation.

## Usage Changes

None.

## Considerations

Same as in #2135, the new guide makes use of new WordPress functionality in 5.3. Adding backwards compatibility could make the guide more complicated. Let’s discuss in #2135 how we should proceed about the backwards compatibility.

## Testing

None.

### CODE REVIEW

File: docs/guides/date-time.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+## Creating dates from strings
+
+If you want to create a `DateTime` object or a timestamp from a time string, you have multiple possibilities. What you need to use depends on how you stored your dates and how you use them.
+
+WordPress recommends to **store your dates either as Unix timestamps or formats that are precise moments in time**, such as [`DATE_RFC3339`](https://www.php.net/manual/en/class.datetimeinterface.php#datetime.constants.rfc3339)/`DATE_ATOM`.
+
+### Create a date from a timestamp
+
+When you have a **timestamp**, you can create your object with `DateTime` or `date_create()`.
```
[the remark this answers is not in the document]
Comment (gchtr) [reply docs/guides/date-time.md]: Cool, thanks for diving into the guide 👍🤩.

I updated the guide with immutable objects and functions in https://github.com/timber/timber/pull/2136/commits/ae32ada5357a1f9cf6d6412ef814f4faab30a8f9.

File: docs/guides/date-time.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+$timestamp = $datetime->getTimestamp();
+```
+
+If you don’t know the exact format of the date, you can try using `strtotime()` or `date_create()`. Valid formats are explained in [Supported Date and Time Formats](https://www.php.net/manual/en/datetime.formats.php).
+
+```php
+$timestamp = strtotime( '2008-08-07 18:11:31' );
+
+$datetime = date_create( '2008-08-07 18:11:31' );
```
Comment (acobster) [inline docs/guides/date-time.md]: missed a couple `date_time_immutable`s here :)

Comment (gchtr) [reply docs/guides/date-time.md]: Darn! Fixed.

**Comment by gchtr** [comment]
We’ll have to wait for #2135 to be merged before this can be merged.

**Comment by jarednova** [comment]
@gchtr could we switch the base branch on this to be `2.x-twig-date-compat` to join into #2135, or is it easier to make it direct to 2.x?

**Comment by gchtr** [comment]
> @gchtr could we switch the base branch on this to be `2.x-twig-date-compat` to join into #2135, or is it easier to make it direct to 2.x?

I had that before, but somehow I couldn’t update the branch, it always said «Not found» 🤷‍♂️. No idea why… but anyways, let’s try to do this manually then.

**Pull request merged**